pkg/corerp/api/v20231001preview/environment_conversion.go

```diff
@@ -180,6 +183,59 @@ func (dst *EnvironmentResource) ConvertFrom(src v1.DataModelInterface) error {
 	return nil
 }
 
+func toRecipeConfigDatamodel(config *RecipeConfigProperties) datamodel.RecipeConfigProperties {
+	if config != nil {
+		recipeConfig := datamodel.RecipeConfigProperties{}
+		if config.Terraform != nil {
+			recipeConfig.Terraform = datamodel.TerraformConfigProperties{}
+			if config.Terraform.Authentication != nil {
+				recipeConfig.Terraform.Authentication = datamodel.AuthConfig{}
+				gitConfig := config.Terraform.Authentication.Git
+				if gitConfig != nil {
+					recipeConfig.Terraform.Authentication.Git = datamodel.GitAuthConfig{}
+					if gitConfig.Pat != nil {
+						p := map[string]datamodel.SecretConfig{}
+						for k, v := range gitConfig.Pat {
+							p[k] = datamodel.SecretConfig{
+								Secret: to.String(v.Secret),
+							}
+						}
+						recipeConfig.Terraform.Authentication.Git.PAT = p
+					}
+
+				}
+			}
+		}
+		return recipeConfig
+	}
+	return datamodel.RecipeConfigProperties{}
+}
+
+func fromRecipeConfigDatamodel(config datamodel.RecipeConfigProperties) *RecipeConfigProperties {
+	if !reflect.DeepEqual(config, datamodel.RecipeConfigProperties{}) {
+		recipeConfig := &RecipeConfigProperties{}
+		if !reflect.DeepEqual(config.Terraform, datamodel.TerraformConfigProperties{}) {
```

Comment:
Do we need to use reflection? Can this be `if config.Terraform != (datamodel.TerraformConfigProperties{})` instead for example?

Reply:
that kept throwing error we cannot do that, so I had to use reflect.
I dont know if i am missing something, i spent lot of time to get it working the way shown in example or the way we do it for providers but i kept getting the same error